e2e/e2e.go

```diff
@@ -192,6 +197,70 @@ func main() {
 
 	log.Logger = log.Output(zerolog.ConsoleWriter{Out: os.Stderr})
 
+	// runEndToEndTests is a flag that denotes whether or not this needs to actually
+	// run or not based off of the filepath.Walk function immediately proceeding.
+	var runEndToEndTests bool
+
+	err := filepath.Walk(".", func(path string, info os.FileInfo, err error) error {
+		if runEndToEndTests {
+			// No need to keep walking through files if we've already found one file
+			// that requries e2e tests.
+			return nil
+		}
+
+		if err != nil {
+			return err
+		}
+
+		if info.Mode()&os.ModeSymlink == os.ModeSymlink {
+			// Skip symlinks.
+			return nil
+		}
+
+		if !strings.HasSuffix(path, "_test.go") {
+			// Skip all files that aren't go test files.
+			return nil
+		}
+
+		f, err := os.ReadFile(path)
```

Comment:
lol guess who forgot to close the file again

Reply:
wait i don't have to in this case. played myself.

Reply:
yeah I was about to comment, I don't think reading it into memory is a great idea if we can avoid it